esteid/signing/types.py

```diff
@@ -0,0 +1,103 @@
+import base64
+import os
+from typing import Union
+
+from django.core.files import File
+
+
+class DataFile:
+    """
+    Interface for file objects that are added to signed containers
+
+    The container cares about 3 attributes:
+    * file name (base name only)
+    * mime type
+    * the content, for calculating hash.
+    """
+
+    def __init__(self, wraps: Union[str, File], mime_type: str = None, content: bytes = None, file_name: str = None):
```

Comment:
The point of this wrapper was to make users type as little code as possible. 
I don't think there would be issues with static analyzers, and as for readability,
we could use a pure namedtuple/dataclass as earlier, but then it would require users to explicitly open and read files which surely makes their code less readable :) The `file_name` arg is there for the case where it can not be inferred from `wraps`, like `ContentFile`. I could document this to avoid questions